### PR TITLE
SignalFx JSON parser

### PR DESCRIPTION
**Pull request by keitwb**
Switch from proprietary json format to SignalFx one.

### CODE REVIEW

File: json.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		return ret
+	}
+	panic(fmt.Sprintf("Unknown metric type: %v\n", mt))
+}
+
+func fromTs(ts int64) time.Time {
+	if ts > 0 {
+		return time.Unix(0, ts*time.Millisecond.Nanoseconds())
+	}
+	return time.Now().Add(-time.Duration(time.Millisecond.Nanoseconds() * ts))
```
Comment (jaysfx) [inline json.go]: When is the timestamp negative?

Comment (keitwb) [reply json.go]: I have no idea, I copied this straight from an unexported method in https://github.com/signalfx/gateway/blob/master/protocol/signalfx/signalfx.go#L81

Comment (keitwb) [reply json.go]: I guess at relative timestamps from the present are allowed?

Comment (jaysfx) [reply json.go]: ¯\_(ツ)_/¯

**Review by ghost — APPROVED** [review]

File: json.go (hunk 1 above)
Comment (keitwb) [reply json.go]: They would be accepted just like ingest will accept them.

**Pull request merged**